anvil/src/main/kotlin/dev/inkremental/Anvil.kt

```diff
@@ -0,0 +1,326 @@
+        if (Looper.myLooper() != Looper.getMainLooper()) {
+            synchronized(Anvil::class.java) {
+                if (anvilUIHandler == null) {
+                    anvilUIHandler = Handler(Looper.getMainLooper())
+                }
+            }
+            anvilUIHandler?.removeCallbacksAndMessages(null)
+            anvilUIHandler?.post(anvilRenderRunnable)
+            return
+        }
+        val set: MutableSet<Mount> = mutableSetOf()
+        set.addAll(mounts.values)
+        for (m in set) {
+            render(m)
+        }
+    }
+
+    /**
+     * Mounts a renderable function defining the layout into a View. If host is a
+     * viewgroup it is assumed to be empty, so the Renderable would define what
+     * its child views would be.
+     * @param v a View into which the renderable r will be mounted
+     * @param r a Renderable to mount into a View
+     */
+    fun <T : View> mount(v: T, r: Renderable): T {
+        val m = Mount(v, r)
+        mounts[v] = m
+        render(v)
+        return v
+    }
+
+    /**
+     * Unmounts a  mounted renderable. This would also clean up all the child
+     * views inside the parent ViewGroup, which acted as a mount point.
+     * @param v A mount point to unmount from its View
+     */
+    @JvmOverloads
+    fun unmount(v: View?, removeChildren: Boolean = true) {
+        val m = mounts[v]
+        if (m != null) {
+            mounts.remove(v)
+            if (v is ViewGroup) {
+                val viewGroup = v
+                val childCount = viewGroup.childCount
+                for (i in 0 until childCount) {
+                    unmount(viewGroup.getChildAt(i))
+                }
+                if (removeChildren) {
+                    viewGroup.removeViews(0, childCount)
+                }
+            }
+        }
+    }
+
+    /**
+     * Returns currently rendered Mount point. Must be called from the
+     * Renderable's view() method, otherwise it returns null
+     * @return current mount point
+     */
+    fun currentMount(): Mount? {
+        return currentMount
+    }
+
+    /**
+     * Returns currently rendered View. It allows to access the real view from
+     * inside the Renderable.
+     * @return currently rendered View
+     */
+    fun <T : View?> currentView(): T? {
+        return if (currentMount == null) {
+            null
+        } else currentMount?.iterator?.currentView() as T?
+    }
+
+    fun render(v: View?) {
+        val m = mounts[v] ?: return
+        render(m)
+    }
+
+    fun render(m: Mount) {
+        if (m.lock) {
+            return
+        }
+        m.lock = true
+        val prev = currentMount
+        currentMount = m
+        m.iterator.start()
+        if (m.renderable != null) {
+            m.renderable.view()
+        }
+        m.iterator.end()
+        currentMount = prev
+        m.lock = false
+    }
+
+    //waiting for Kotlin 1.4
```

Comment:
Could you please expand this a bit? You mean SAM interfaces?
What benefit does interface over functional type brings us?

Reply:
Most because of verbosity of usage of Kotlin's  single method interface.

Reply:
And what about pros of `interface Renderable` vs current `() -> Unit`?

Reply:
Yep, managed to remove Renderable interface completely!